### PR TITLE
x-pack/filebeat/input/cel: fix handling of non-200/non-429 status codes

### LINKED ISSUE #33999: [Filebeat] cel input - non HTTP 200 status_code causes rapid retry
When the CEL program returns a `state.status_code` that is non HTTP 200 then the input will repeat the request rapidly. This occurs for status codes that are outside the range that are handled by go-retryablehttp like [201, 499] except 429. If I omit the `status_code` then it basically works as expected by making a request, reporting the error, and then waiting for the next interval. So perhaps when a non-200 `status_code` is reported then the auto-generated `error.message` could include information about the status code and it would wait for the next interval?

In 6.231 sec the input made 10893 requests. That's ~1700 request per second. 😨 

```
% grep -h "HTTP request" cel.trace | wc -l
   10893
% head -1 cel.trace| jq '."@timestamp"'
"2022-12-08T10:38:10.445-0500"
% tail -1 cel.trace| jq '."@timestamp"'
"2022-12-08T10:38:16.676-0500"
```

Version: filebeat version 8.7.0 (arm64), libbeat 8.7.0 [f08eed88d58d33810e6e923c03fad9a3c87bd575 built 2022-12-08 15:32:00 +0000 UTC]

### Reproduction

```go
package main

import (
	"log"
	"net/http"
)

func main() {
	handler := func(writer http.ResponseWriter, request *http.Request) {
		http.Error(writer, "Go away", http.StatusForbidden)
	}
	log.Fatal(http.ListenAndServe("localhost:9888", http.HandlerFunc(handler)))
}
```

```yaml
---

filebeat.inputs:
  - type: cel
    interval: 1m

    resource:
      url: http://localhost:9888
      retry:
        min_wait: 10s
        max_wait: 120s
      tracer:
        filename: cel.trace

    program: |
      get('http://localhost:9888').as(resp, bytes(resp.Body).as(body, {
          "events": (resp.StatusCode == 200) ? try(body.decode_json()) : {"error": {"message": "unexpected status code " + string(resp.StatusCode)} },
          "status_code": resp.StatusCode,
          "want_more": false,
      }))

output.console.pretty: true
```

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

Previously, non-200/non-429 status codes would cause an immediate retry, resulting in spamming the server. Treat these responses as an error and wait until the next event period.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works — manually tested see #33999 for approach
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Fixes #33999

### CODE REVIEW

File: x-pack/filebeat/input/cel/input.go [hunk 1]
```diff
@@ -529,10 +529,14 @@ func handleResponse(log *logp.Logger, state map[string]interface{}, limiter *rat
 					waitUntil = t
 				}
 			}
-			fallthrough
-		default:
-			delete(state, "events")
 			return false, waitUntil, nil
+		default:
```
Comment (efd6) [inline x-pack/filebeat/input/cel/input.go:533]: We should consider stratifying you.

File: x-pack/filebeat/input/cel/input.go [hunk 2]
```diff
@@ -529,10 +529,14 @@ func handleResponse(log *logp.Logger, state map[string]interface{}, limiter *rat
...
+			status := http.StatusText(statusCode)
+			if status == "" {
+				status = "unknown status code"
+			}
+			state["events"] = map[string]interface{}{"error.message": fmt.Sprintf("failed http request with %s: %d", status, statusCode)}
```
Comment (andrewkroh) [inline x-pack/filebeat/input/cel/input.go:538]: The `error.message` will result in a key containing a dot. Can you please change that to produce `{"error":{"message":...`.

**Review by andrewkroh — APPROVED** [review]
LGTM. I tested it. Just a minor issue with the "dot" in the field name.

File: x-pack/filebeat/input/cel/input.go (hunk 2 above)
Comment (efd6) [reply x-pack/filebeat/input/cel/input.go:538]: That should happen in the other `error.message` constructions too then.

File: x-pack/filebeat/input/cel/input.go [hunk 3]
```diff
@@ -853,29 +853,33 @@ func evalWith(ctx context.Context, prg cel.Program, state map[string]interface{}
...
-		state["events"] = map[string]interface{}{"error.message": fmt.Sprintf("failed json conversion: %v", err)}
+		state["events"] = errorMessage(fmt.Sprintf("failed json conversion: %v", err))
 		return state, fmt.Errorf("failed json conversion: %w", err)
 	}
 	return res, nil
 }
 
+func errorMessage(msg string) map[string]interface{} {
+	return map[string]interface{}{"error": map[string]interface{}{"message": msg}}
+}
```
Comment (efd6) [inline x-pack/filebeat/input/cel/input.go:881]: Query whether this should be 
```
func errorMessage(msg ...string) map[string]interface{} {
	return map[string]interface{}{"error": map[string]interface{}{"message": msg}}
}
```

**Pull request merged**